src/nickserv.c

```diff
@@ -708,7 +708,7 @@ static const char *
 visible_email_addr(struct userNode *user, struct handle_info *hi)
 {
     if (hi->email_addr) {
-        if (oper_has_access(user, nickserv, nickserv_conf.email_visible_level, 1)) {
+        if (hi == user->handle_info || oper_has_access(user, nickserv, nickserv_conf.email_visible_level, 1)) {
```

Comment:
That is still not a use case for seeing the information.  If a user needs a password reset, or whatever, they can request that.  If they're about to lose an email address, and don't remember what is using it, they have to interact with everything that might be tied to that email address and potentially change them -- so why not just try to change their GS/srvx account to the address they want to use?